lua/luaUtils.lua

```diff
@@ -0,0 +1,83 @@
+function getMacroValue (options)
+    --[[
+        Gets the value of a macro from the environment, or returns the given default value
+        Signature:
+            getMacroValue{macro="macro_name", default="default_value"}
+        Args:
+            macro: String, the macro to look up in the environment
+            default: String, the value to return if the macro if the macro is not set in the environment
+        Raises:
+            Error if the macro could not be found in the environment and no default is given
+    ]]
+
+    if type(options.macro) ~= "string" then
+        error("No macro given")
+    end
+    macro = options.macro
+    default = options.default
+
+    macroFromEnvironment = os.getenv(macro)
+
+    if (macroFromEnvironment~=nil) then
+        -- Macro set to value from environment
+        macroValue = macroFromEnvironment
+        print(string.format("%s set from environment as %s", macro, macroFromEnvironment))
+    elseif (type(default) == "string") then
+        -- No macro environment variable set, use default
+        macroValue = default
+    else
+        error(string.format("Macro %s not in enviroment with no default value", macro))
+    end
+
+    return macroValue
+end
+
+function setAsynOptions(device, port, baud, bits, parity, stop)
+    --[[
+        Sets the Asyn Options
+        Args:
+            device: String, the name of the asyn port
+            port: String, the name of the physical port the device is connected to
+            baud: Integer, The baud rate of the device
+            bits: Integer, the number of data bits
+            parity: String, the device parity
+            stop: Integer, the number of stop bits.
+    ]]
+    iocsh.drvAsynSerialPortConfigure(device, port, 0, 0, 0, 0)
+    iocsh.asynSetOption(device, -1, "baud", baud)
+    iocsh.asynSetOption(device, -1, "bits", bits)
+    iocsh.asynSetOption(device, -1, "parity", parity)
+    iocsh.asynSetOption(device, -1, "stop", stop)
+end
+
```

Comment:
In asyn, port is usually the asyn port name - do we want to swap device and port naming?